src/api/web-extension.js

```diff
@@ -174,4 +168,28 @@ export default {
 
     return true;
   },
+
+  /**
+   * Handle messages coming from the background script.
+   *
+   * @param {Object} message
+   *        The incoming message, with the following structure:
+   * ```js
+   * {
+   *  type: "message-type",
+   *  data: { ... },
+   * }
+   * ```
+   */
+  async _handleMessage(message) {
+    switch (message.type) {
+      case "update-state": {
+        // TODO: Update the UI accordingly.
+        console.log('Received new studyes! ' + message.data);
```

Comment:
@hamilton this is your hook in point to update the UI!  See the docstrings for `_sendStateUpdateToUI` for the format of the `message.data` field for this message.